services/document-manager/backend/app/docman/utils/document_upload_helper.py

```diff
@@ -87,36 +89,86 @@ def initiate_document_upload(cls, document_guid, file_path, folder, file_size, v
 
-        response.headers['Tus-Resumable'] = TUS_API_VERSION
-        response.headers['Tus-Version'] = TUS_API_SUPPORTED_VERSIONS
         response.headers['Location'] = f'{Config.DOCUMENT_MANAGER_URL}/documents/{document_guid}'
         response.headers['Upload-Offset'] = 0
         response.headers['Upload-Expires'] = upload_expiry
-        response.headers[
-            'Access-Control-Expose-Headers'] = 'Tus-Resumable,Tus-Version,Location,Upload-Offset,Upload-Expires,Content-Type'
+
         if version_guid is not None:
             response.headers['Document-Version'] = version_guid
-            response.headers[
-                'Access-Control-Expose-Headers'] = response.headers[
-                'Access-Control-Expose-Headers'] + ',Document-Version'
+
+            if not is_s3_multipart:
+                response.headers[
+                    'Access-Control-Expose-Headers'] = response.headers[
+                    'Access-Control-Expose-Headers'] + ',Document-Version'
 
         response.autocorrect_location_header = False
 
-        return response, object_store_path
+        return response, object_store_path, multipart_upload_path, s3_upload['uploadId'] if s3_upload is not None else None
+
+    @classmethod
+    def _initialize_tusd_upload(cls, document_guid, headers):
+        resp = None
+        try:
+            resp = requests.post(url=Config.TUSD_URL,
+                                     headers=headers, data=request.data)
+        except Exception as e:
+            message = f'POST request to object store raised an exception:\n{e}'
+            current_app.logger.error(message)
+            raise InternalServerError(message)
+
+            # Validate the request
+        if resp.status_code != requests.codes.created:
+            message = f'POST request to object store failed: {resp.status_code} ({resp.reason}): {resp._content}'
+            current_app.logger.error(
+                    f'POST resp.request:\n{resp.request.__dict__}')
+            current_app.logger.error(f'POST resp:\n{resp.__dict__}')
+            current_app.logger.error(message)
+            if resp.status_code == requests.codes.not_found:
+                raise NotFound(message)
+            raise BadGateway(message)
+
+            # Set object store upload data in cache
+        object_store_upload_resource = urlparse(
+                resp.headers['Location']).path.split('/')[-1]
+        object_store_path = Config.S3_PREFIX + \
+                object_store_upload_resource.split('+')[0]
+        cache.set(
+                OBJECT_STORE_UPLOAD_RESOURCE(
+                    document_guid), object_store_upload_resource,
+                CACHE_TIMEOUT)
+        cache.set(OBJECT_STORE_PATH(document_guid),
+                      object_store_path, CACHE_TIMEOUT)
+                  
+        return object_store_path
 
     @ classmethod
```

Comment:
not anything you did here but I don't think there should be a space in `@classmethod`?